glusterd2/commands/volumes/volfiles.go

```diff
@@ -33,3 +37,38 @@ func volfilesListHandler(w http.ResponseWriter, r *http.Request) {
 	}
 	restutils.SendHTTPResponse(ctx, w, http.StatusOK, volfiles)
 }
+
+func volfilesGetHandler(w http.ResponseWriter, r *http.Request) {
+	ctx := r.Context()
+
+	volfiles, err := volgen.GetVolfiles()
+
+	if err != nil {
+		restutils.SendHTTPError(ctx, w, http.StatusInternalServerError, "unable to get list of volfiles")
+		return
+	}
+
+	restutils.SendHTTPResponse(ctx, w, http.StatusOK, volfiles)
+}
+
+func volfileGetHandler(w http.ResponseWriter, r *http.Request) {
+	ctx := r.Context()
+	volfile := mux.Vars(r)["volfileid"]
+
+	//remove extra slash in last ( as StrictSlash is false in mux.Route)
+	volfile = strings.TrimSuffix(volfile, "/")
+	volfiles, err := volgen.GetVolfile(volfile)
+
+	if err != nil {
+		if err == errors.ErrVolFileNotFound {
+			restutils.SendHTTPError(ctx, w, http.StatusNotFound, errors.ErrVolFileNotFound)
+		} else {
+			restutils.SendHTTPError(ctx, w, http.StatusInternalServerError, fmt.Sprintf("unable to fetch volfile content %s", volfile))
+		}
+		return
+	}
+
+	restutils.SendHTTPResponse(ctx, w, http.StatusOK, nil)
```

Comment:
volfiles can be returned here itself

Reply:
cannot return volfiles here, as volfiles are slice of bytes. so way passing nil to set some header for response

Reply:
This isn't clean. All our APIs have JSON responses. Alternatively, you can return content as json. You can use a struct such as this:

```go
type VolfileResp struct {
    ID string
    Content string
}
```

Reply:
This cannot be done now, current volfile content already present in ETCD as a string (which contains newline characters) if it is presented as a string it won't be in human-readable form.